### PR TITLE
upgrade to react 18 and enable the new concurrent renderer

### PR DESCRIPTION
**Pull request by KKoukiou**
https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-client-rendering-apis

### CODE REVIEW

File: pkg/lib/cockpit-components-dialog.jsx [hunk 1]
```diff
@@ -271,12 +271,14 @@ export function show_modal_dialog(props, footerProps) {
     const dialogName = 'cockpit_modal_dialog';
     // don't allow nested dialogs, just close whatever is open
     const curElement = document.getElementById(dialogName);
+    const current = createRoot(curElement);
     if (curElement) {
-        ReactDOM.unmountComponentAtNode(curElement);
+        current.unmount();
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-dialog.jsx:276]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17939-20221121-134727-dbe868f6-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/lib/cockpit-components-dialog.jsx [hunk 2]
```diff
@@ -285,7 +287,7 @@
     const closeCallback = function() {
         if (origCallback)
             origCallback.apply(this, arguments);
-        ReactDOM.unmountComponentAtNode(rootElement);
+        root.unmount();
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-dialog.jsx:290]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17939-20221121-134727-dbe868f6-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/lib/cockpit-components-dialog.jsx [hunk 3]
```diff
@@ -299,7 +301,7 @@
         // the input focus from whatever element has it, which is
         // unpleasant and also disrupts the tests.
         if (rootElement.offsetParent)
-            ReactDOM.render(<Dialog {...dialogObj.props} />, rootElement);
+            root.render(<Dialog {...dialogObj.props} />);
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-dialog.jsx:304]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17939-20221121-134727-dbe868f6-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by jelly** [comment]
As we have coverage now, we have a development build we run tests against so that might be useful to look at so you don't have to run all the tests locally:

https://cockpit-logs.us-east-1.linodeobjects.com/pull-17939-20221121-134727-dbe868f6-fedora-36-devel/log.html

For example:

```
  File "/work/bots/make-checkout-workdir/test/common/cdp.py", line 232, in command
    raise RuntimeError(res["error"])
RuntimeError: Error: createRoot(...): Target container is not a DOM element.

# Result testKeepKeys (__main__.TestStorageLuks.testKeepKeys) failed
```

**Comment by KKoukiou** [comment]
This changes so many pages which starts the 3 times retry in the tests and brings all flaky on the surface. Uff. Removed some super flaky pages and will try to add them in followups.

**Comment by marusak** [comment]
Looks good, but it shows you commited pixel tests with 0 changes? Can you explain what is happening there?

**Comment by KKoukiou** [comment]
> Looks good, but it shows you commited pixel tests with 0 changes? Can you explain what is happening there?



> Looks good, but it shows you commited pixel tests with 0 changes? Can you explain what is happening there?

OH good catch. Indeed I had updated one pixel test, but then I reverted the change.

**Review by jelly — APPROVED** [review]

**Comment by jelly** [comment]
Approved if you fix the empty test/reference update :)

**Pull request merged**